packages/core/data-transfer/src/strapi/providers/utils.ts

```diff
@@ -34,26 +34,40 @@ export const createDispatcher = (ws: WebSocket) => {
     return new Promise<U | null>((resolve, reject) => {
       const uuid = randomUUID();
       const payload = { ...message, uuid };
+      let numberOfTimesMessageWasSent = 0;
 
       if (options.attachTransfer) {
         Object.assign(payload, { transferID: state.transfer?.id });
       }
 
       const stringifiedPayload = JSON.stringify(payload);
-
       ws.send(stringifiedPayload, (error) => {
         if (error) {
           reject(error);
         }
       });
 
+      const sendPeriodically = () => {
+        if (numberOfTimesMessageWasSent < 5) {
+          numberOfTimesMessageWasSent += 1;
+          ws.send(stringifiedPayload, (error) => {
+            if (error) {
+              reject(error);
+            }
+          });
+        } else {
+          reject(new ProviderError('error', 'Request timed out'));
+        }
+      };
+      const interval = setInterval(sendPeriodically, 30000);
```

Comment:
I think 30s is too long. I forget if we have a max size for a message, but they shouldn't be very large should they? I think 10s should be fine, but we should do some testing.

Reply:
yeah we can reduce, but 30s does work I did test it intensively but maybe we can drop it to half or something should be fine, I just don't want it to execute to quickly either

Reply:
> yeah we can reduce, but 30s does work I did test it intensively but maybe we can drop it to half or something should be fine, I just don't want it too execute to quickly either

What happens if two messages get sent and both are actually received?